pkg/graveler/graveler.go

```diff
@@ -1286,6 +1295,66 @@ func (g *KVGraveler) SaveGarbageCollectionCommits(ctx context.Context, repositor
 	}, err
 }
 
+type GCUncommittedMark struct {
+	BranchID BranchID `json:"branch"`
+	Key      Key      `json:"key"`
+}
+
+func (g *KVGraveler) GetGarbageCollectionUncommitted(ctx context.Context, repository *RepositoryRecord, mark GCUncommittedMark) ([]*ValueRecord, *GCUncommittedMark, error) {
+	var entries []*ValueRecord
+	bItr, err := g.ListBranches(ctx, repository)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer bItr.Close()
+	bItr.SeekGE(mark.BranchID)
+
+	count := 0
+	for bItr.Next() {
+		b := bItr.Value()
+		itr, err := g.listStagingArea(ctx, b.Branch)
+		if err != nil {
+			bItr.Close()
+			return nil, nil, err
+		}
+		itr.SeekGE(mark.Key)
+		for itr.Next() {
+			count += 1
+			entry := itr.Value()
+			if count > GCMaxUncommittedObjects {
+				contToken := GCUncommittedMark{
+					BranchID: b.BranchID,
+					Key:      entry.Key,
+				}
+				return entries, &contToken, nil
+			}
+			entries = append(entries, entry)
+			itr.Close()
+			if itr.Err() != nil {
+				return nil, nil, itr.Err()
+			}
+		}
+		itr.Close()
+	}
+	return entries, nil, bItr.Err()
+}
+
+func (g *KVGraveler) SaveGarbageCollectionUncommitted(ctx context.Context, repository *RepositoryRecord, filename, runID string) (*GarbageCollectionRunMetadata, error) {
+	err := g.garbageCollectionManager.SaveGarbageCollectionUncommitted(ctx, repository, filename, runID)
+	if err != nil {
+		return nil, err
+	}
+	uncommittedLocation, err := g.garbageCollectionManager.GetUncommittedLocation(runID, repository.StorageNamespace)
```

Comment:
Can't `uncommittedLocation ` be the output of `SaveGarbageCollectionUncommitted` and remove `GetUncommittedLocation` from the API?

Reply:
GetUncommittedLocation would be used by the client to get the the previous run's uncommitted data